helm/promtail/templates/service.yaml

```diff
@@ -9,7 +9,7 @@ metadata:
   {{- end }}
   labels:
     {{- include "promtail.selectorLabels" . | nindent 8 }}
-    {{- with .Values.serviceLabels }}
+    {{- with .Values.promtail.serviceLabels }}
```

Comment:
I think you're creating a custom service for monitoring here.
Please make sure:
- it has a meaningful name so we clearly see it's a GS-specific service
- it has the monitoring label so that it gets scraped

Reply:
> it has the monitoring label so that it gets scraped

As mentioned in the PR's first comment, promtail is scraped by prometheus and thus have the monitoring label : ```NAME       TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE   LABELS
promtail   ClusterIP   None         <none>        9091/TCP   24d   app.kubernetes.io/instance=promtail,app.kubernetes.io/managed-by=Helm,app.kubernetes.io/name=promtail,giantswarm.io/monitoring=true```

Concerning the service's name do you have any suggestion ?

Reply:
> As mentioned in the PR's first comment, promtail is scraped by prometheus

ooh, yes, because values contains this:
```
  serviceLabels: {
    giantswarm.io/monitoring: "true"
  }
```
although it only applies to our monitoring service, as far as I can see.

Does it make sense to have it in `values.yaml` if it's only for our monitoring service?
Does it make sense to have it under `.promtail` if it's specific to our custom chart?



> Concerning the service's name do you have any suggestion ?

If we take loki as example (https://github.com/giantswarm/loki-app/blob/master/helm/loki/templates/metrics.yaml), that would be `promtail-metrics`.
But it looks like there's already such a service upstream (https://github.com/grafana/helm-charts/blob/main/charts/promtail/templates/service-metrics.yaml).
So `promtail-giantswarm-monitoring`?

BTW, This service should have a flag in values so we can enable/disable it, and default values should disable servicemonitor.
Then when we have servicemonitor support all around we can switch from annotation to servicemonitor by updating values.

Reply:
Why not using the upstream service if there's one ?

Reply:
> Why not using the upstream service if there's one ?

The issue I had with the existing service was that for it to be created, one has to enable the servicemonitor section in the `promtail` section in the values and I thought it would be easier to have a custom service not relying on this section as long as we don't rely on servicemonitors not to confuse ourselves. 

Concerning @hervenicol comment, I moved service related values outside of the `promtail` section in `values.yaml` and set its name to `promtail-giantswarm-monitoring`. I let the labels in the values file as hardcoding it in the service brings out an error. 
The service is enable if the servicemonitor in the `promtail` section of the values file is disabled. 

Are you ok with this config ?

Reply:
> Why not using the upstream service if there's one ?

There is an upstream service that is named `promtail-metrics`, true. It has the same name we would have given it, but it does fulfil the purpose we are looking for.

Here, we want to provide a service that will have the `giantswarm.io/monitoring` annotation, so that our monitoring stack scrapes promtail's metrics.
This is a workaround because we can't set the right annotation to the specific services in the upstream chart, so we create a new service dedicated to that.
The clean solution would be to use a servicemonitor, but this is not yet supported on all of our clusters.

So, the requirements for this service are:
- to have the `giantswarm.io/monitoring`
- expose the port for promtail metrics

This is why I would hardcode the annotation on it, and remove it from `values`.

Reply:
> I let the labels in the values file as hardcoding it in the service brings out an error.

Because the indentation does not match.
Using `nindent 4` instead of `nindent 8` should fix.

Reply:
Indeed ! It now works :)